tests/test_performance_curve.py

```diff
@@ -1,62 +1,112 @@
+    np.testing.assert_array_equal(x1.case_counts, case_counts)
+
+    np.testing.assert_array_almost_equal(x2.performance_values, np.array([0.333, 0.666, 0.666, 1., 1.]), decimal=3)
+    np.testing.assert_array_equal(x2.case_counts, case_counts)
+
+    np.testing.assert_array_equal(x3.performance_values, np.array([1., 0.75, 0.6]))
+    np.testing.assert_array_equal(x3.case_counts, np.array([2, 4, 5]))
+
+    np.testing.assert_array_equal(x4.performance_values, np.array([1., 1.]))
+    np.testing.assert_array_equal(x4.case_counts, np.array([3, 5]))
+
+    # Given two label/score arrays of different lengths
+    y_true = np.array([1, 0, 1])
     y_score = np.array([0.1, 0.4, 0.91, 0.8, 0.05])
 
-    #
-    # Whens
-    #
-
-    # measuring the performance curve of all data points
-    x1 = performance_curve(y_true, y_score, precision_score)
-    x2 = performance_curve(y_true, y_score, recall_score)
-    x3 = performance_curve(y_true, y_score, accuracy_score)
-    x4 = performance_curve(y_true, y_score, roc_auc_score)
-
-    #
-    # Thens
-    #
-    sorted_y_true = np.array([0.91, 0.8, 0.4, 0.1, 0.05])
-    np.testing.assert_array_equal(x1.cutoff, sorted_y_true)
-    np.testing.assert_array_equal(x2.cutoff, sorted_y_true)
-    np.testing.assert_array_equal(x3.cutoff, sorted_y_true)
-    np.testing.assert_array_equal(x4.cutoff, sorted_y_true)
-
-    np.testing.assert_array_almost_equal(x1.metric_value, np.array([1., 1., 0.666, 0.5, 0.4]), decimal=3)
-    np.testing.assert_array_almost_equal(x2.metric_value, np.array([0.5, 1., 1., 1., 1.]), decimal=3)
-    np.testing.assert_array_almost_equal(x3.metric_value, np.array([0.8, 1, 0.8, 0.6, 0.4]), decimal=3)
-    np.testing.assert_array_almost_equal(x4.metric_value, np.array([0.75, 1, 0.833, 0.666, 0.5]), decimal=3)
-
-    #
-    # Whens
-    #
-
-    # dividing the data points into 3 bins and measuring the performance curve
-    x1 = performance_curve(y_true, y_score, precision_score, num_bins=3)
-    x2 = performance_curve(y_true, y_score, recall_score, num_bins=3)
-    x3 = performance_curve(y_true, y_score, accuracy_score, num_bins=3)
-    x4 = performance_curve(y_true, y_score, roc_auc_score, num_bins=3)
-
-    #
-    # Thens
-    #
-    sorted_y_true_ub = np.array([0.91, 0.4, 0.05])
-    np.testing.assert_array_equal(x1.cutoff, sorted_y_true_ub)
-    np.testing.assert_array_equal(x2.cutoff, sorted_y_true_ub)
-    np.testing.assert_array_equal(x3.cutoff, sorted_y_true_ub)
-    np.testing.assert_array_equal(x4.cutoff, sorted_y_true_ub)
-
-    np.testing.assert_array_almost_equal(x1.metric_value, np.array([1., 0.5, 0.4]), decimal=3)
-    np.testing.assert_array_almost_equal(x2.metric_value, np.array([1., 1., 1.]), decimal=3)
-    np.testing.assert_array_almost_equal(x3.metric_value, np.array([1., 0.6, 0.4]), decimal=3)
-    np.testing.assert_array_almost_equal(x4.metric_value, np.array([1., 0.666, 0.5]), decimal=3)
+    # Expect to raise error
+    with np.testing.assert_raises(AssertionError):
```

Comment:
Interesting choice to use numpy testing utilities for this. I didn't know they provided this functionality. No need to make a change, but I'm curious if there is a reason to use this instead of [the same functionality provided by `pytest`](https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions). 

```python
with pytest.raises(AssertionError):
    ...
```

Reply:
The main reason is just because my PyCharm keeps complaining that I need to include `pytest` in the project requirements and I think we agreed that `requirements.txt` should only include necessary packages for the users (not developers). I know that we already included `pytest` in `tox.ini`, but not sure why my PyCharm still complains. Anyway, since we already use `np.testing` for all the test cases above and there's not any significant difference between the two ways of raising exceptions, I figure it would be safe to use `np.testing.assert_raises()` in this case.